### PR TITLE
Add `LiquibasePreparer.forFile`

### PR DESCRIPTION
**Pull request by remoun**
Like `forClasspathLocation` but with a `FileSystemResourceAccessor`

### CODE REVIEW

**Comment by remoun** [comment]
@tomix26 what do you think of this change?

**Comment by tomix26** [comment]
Thanks for your contribution. I'll have to look at it first and verify the changes. I'll do it as soon as I can, I promise.

**Comment by tomix26** [comment]
There are some compilation errors, so you can check them out and fix it. Thanks in advance!

File: src/main/java/io/zonky/test/db/postgres/embedded/LiquibasePreparer.java [hunk 1]
```diff
@@ -30,17 +32,26 @@
...
     public static LiquibasePreparer forClasspathLocation(String location, Contexts contexts) {
-        return new LiquibasePreparer(location, contexts);
+        return new LiquibasePreparer(location, new ClassLoaderResourceAccessor(), contexts);
+    }
+    
+    public static LiquibasePreparer forFile(File file) {
+        return forFile(file, new Contexts());
+    }
+    public static LiquibasePreparer forFile(File file, Contexts contexts) {
+        return LiquibasePreparer(file.getName(), new FileSystemResourceAccessor(file.getParentFile()), contexts);
```
Comment (tomix26) [inline src/main/java/io/zonky/test/db/postgres/embedded/LiquibasePreparer.java:49]: There should be a check for NPE.

File: src/main/java/io/zonky/test/db/postgres/embedded/LiquibasePreparer.java [hunk 2]
```diff
@@ -50,8 +61,14 @@ public void prepare(DataSource ds) throws SQLException {
         try {
             connection = ds.getConnection();
             Database database = getInstance().findCorrectDatabaseImplementation(new JdbcConnection(connection));
-            Liquibase liquibase = new Liquibase(location, new ClassLoaderResourceAccessor(), database);
+
+            Liquibase liquibase = new Liquibase(location, accessor, database);
             liquibase.update(contexts);
+            
+            // Try and commit the changes.
+            connection.commit();
```
Comment (tomix26) [inline src/main/java/io/zonky/test/db/postgres/embedded/LiquibasePreparer.java:69]: Is this change necessary? I would prefer to keep it as it was. I guess the calling of `liquibase.update(contexts)` should commit all the changes, right? And the closing logic is already implemented in the finally block.

Comment (remoun) [reply src/main/java/io/zonky/test/db/postgres/embedded/LiquibasePreparer.java:69]: Unconditionally rolling back in the `finally` seems incorrect. Closing without committing seems risky, so I added commit + close + set to null

Comment (tomix26) [reply src/main/java/io/zonky/test/db/postgres/embedded/LiquibasePreparer.java]: You're right that the previous rolling back logic was incorrect. I've replaced it with a try-with-resources block.

File: src/test/java/io/zonky/test/db/postgres/junit/LiquibasePreparerFileContextTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import org.junit.Test;
+
+import java.io.File;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+import static org.junit.Assert.assertEquals;
+
+public class LiquibasePreparerFileContextTest {
```
Comment (tomix26) [inline src/test/java/io/zonky/test/db/postgres/junit/LiquibasePreparerFileContextTest.java:28]: Thanks for the tests. Now it's much better 👍

**Comment by tomix26** [comment]
Thanks again for the pull request. I've just tweaked a few details and squashed the commits into a single one for better clarity. And I'm going to release it in the next version.
